test-mocha/prettier/sort-lit-get-properties.test.js

```diff
@@ -0,0 +1,123 @@
+          return {
+           ...super.properties,
+            _myPrivA: { type: String },
+            d: { type: String },
+            a: { type: String },
+            b: { type: String },
+            e: { type: String },
+            c: { type: String },
+            _myPrivB: { type: String },
+          };
+        }
+      }
+      `;
+
+    const validCode = `class MyComponent extends LitElement {
+  static get properties() {
+    return {
+      ...super.properties,
+      a: { type: String },
+      b: { type: String },
+      c: { type: String },
+      d: { type: String },
+      e: { type: String },
+      _myPrivA: { type: String },
+      _myPrivB: { type: String },
+    };
+  }
+}
+`;
+    const formattedCode = await formatPrettierCode(code);
+    expect(formattedCode).to.equal(validCode);
+  });
+
+  it('should move the `...super.properties` to the top', async () => {
+    const code = `
+    class MyComponent extends LitElement {
+      static get properties() {
+          return {
+            _myPrivA: { type: String },
+            d: { type: String },
+            a: { type: String },
+            b: { type: String },
+            ...super.properties,
+            e: { type: String },
+            c: { type: String },
+            _myPrivB: { type: String },
+          };
+      }
+    }
+    `;
+
+    const validCode = `class MyComponent extends LitElement {
+  static get properties() {
+    return {
+      ...super.properties,
+      a: { type: String },
+      b: { type: String },
+      c: { type: String },
+      d: { type: String },
+      e: { type: String },
+      _myPrivA: { type: String },
+      _myPrivB: { type: String },
+    };
+  }
+}
+`;
+    const formattedCode = await formatPrettierCode(code);
+    expect(formattedCode).to.equal(validCode);
+  });
+
+  it('should not order if there is an unknown SpreadElement', async () => {
+    const code = `
+    class MyComponent extends LitElement {
+      static get properties() {
+          return {
+            _myPrivA: { type: String },
+            d: { type: String },
+            a: { type: String },
+            b: { type: String },
+            e: { type: String },
+            c: { type: String },
+            _myPrivB: { type: String },
+            ...foobar,
+          };
+      }
+    }
+    `;
+
+    const validCode = `class MyComponent extends LitElement {
+  static get properties() {
+    return {
+      _myPrivA: { type: String },
+      d: { type: String },
+      a: { type: String },
+      b: { type: String },
+      e: { type: String },
+      c: { type: String },
+      _myPrivB: { type: String },
+      ...foobar,
+    };
+  }
+}
+`;
+    const formattedCode = await formatPrettierCode(code);
+    expect(formattedCode).to.equal(validCode);
+  });
+});
+
+async function formatPrettierCode(code) {
+  return await format(code, {
+    filepath: 'foo.js',
+    plugins: ['./prettier-rules/sort-lit-get-properties.js'],
+  });
+}
```

Comment:
nitpick: there's no need to use async/await in this example as you await just before returning

Adding some JSDoc to indicate that the function returns a promise would be nice.

```suggestion
function formatPrettierCode(code) {
  return format(code, {
    filepath: 'foo.js',
    plugins: ['./prettier-rules/sort-lit-get-properties.js'],
  });
}
```

Reply:
Indeed, I'll do that. :+1: